site/en/blog/renderingng-architecture/index.md

```diff
@@ -0,0 +1,502 @@
+If layout, pre-paint, and paint can be skipped for visual effects,
+they can be run entirely on the compositor thread and skip the main thread.
+
+Browser UI rendering is not depicted directly here,
+but can be thought of as a simplified version of this same pipeline
+(and in fact its implementation shares much of the code).
+Video (also not directly depicted)
+generally renders via independent code that decodes frames into GPU texture tiles
+that are then plugged into compositor frames and the draw step.
+
+## Process and thread structure
+
+{% Aside %}
+There are more CPU processes and threads in the Chromium architecture than pictured here.
+These diagrams focus only on those critical to rendering.
+{% endAside %}
+
+### CPU processes
+
+The use of multiple CPU processes achieves performance and security isolation
+between sites and from browser state,
+and stability and security isolation from GPU hardware.
+
+{% Img
+src="image/ZDZVuXt6QqfXtxkpXcPGfnygYjd2/i2t9ihLrkRg5qxUkXOsd.jpeg",
+alt="Diagram of the various parts of the CPU processes", width="800", height="626" %}
+
+- The **render process** renders, animates, scrolls, and routes input for a single site and tab combination.
+There are many render processes.
+- The **browser process** renders, animates, and routes input for the browser UI
+(including the URL bar, tab titles and icons),
+and routes all remaining input to the appropriate render process.
+There is exactly one browser process.
+- The **Viz process** aggregates compositing from multiple render processes
+plus the browser process.
+It rasters and draws using the GPU. There is exactly one Viz process.
+
+Different [sites](https://web.dev/same-site-same-origin/) always end up in different render processes.
+(In reality,
+[always on desktop; when possible on mobile](https://blog.chromium.org/2019/10/recent-site-isolation-improvements.html).
+I'll write "always" below,
+but this caveat applies throughout.)
+
+Multiple browser tabs or windows of the same site usually go in different render processes,
+unless the tabs are related (one [opening](https://developer.mozilla.org/en-US/docs/Web/API/Window/opener)
+the other).
+Under strong memory pressure on desktop Chromium may put multiple tabs
+from the same site into the same render process even if not related.
+
+Within a single browser tab,
+frames from different sites are always in different render processes from each other,
+but frames from the same site are always in the same render process.
+From the perspective of rendering,
+the important advantage of multiple render processes is that cross-site iframes
+and tabs achieve [performance isolation](/blog/renderingng/#performance-isolation)
+from each other.
+In addition, origins can opt into [even more isolation](https://web.dev/origin-agent-cluster/).
+
+There is exactly one Viz process for all of Chromium.
+After all, there is usually only one GPU and screen to draw to.
```

Comment:
It seemed unnecessary. That may be due to me being British, but the style guide agrees with me https://developers.google.com/style/exclamation-points

Reply:
I wanted it to indicate personality/emotion/excitement, because this is not just code documentation but a blog post. WDYT, is there a requirement somewhere that developers.chrome.com downplay the voice of the writer more so than web.dev or the chromium blog?